src/public/javascripts/ssoAuthES6.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See full license in root of repo. 
+ *
+ * This file shows how to use the SSO API to get a bootstrap token.
+ */
+
+Office.onReady(function(info) {
+    $(document).ready(function() {
+        $('#getGraphDataButton').click(getGraphData);
+    });
+});
+
+let retryGetAccessToken = 0;
+
+async function getGraphData() {
+    try {
+        let bootstrapToken = await OfficeRuntime.auth.getAccessToken({ allowSignInPrompt: true }); 
+        let exchangeResponse = await getGraphToken(bootstrapToken);
+        if (exchangeResponse.claims) {
+            // Microsoft Graph requires an additional form of authentication. Have the Office host 
+            // get a new token using the Claims string, which tells AAD to prompt the user for all 
+            // required forms of authentication.
+            let mfaBootstrapToken = await OfficeRuntime.auth.getAccessToken({ authChallenge: exchangeResponse.claims });
+            exchangeResponse = await getGraphToken(mfaBootstrapToken);
+        }
+        
+        if (exchangeResponse.error) {
+            // AAD errors are returned to the client with HTTP code 200, so they do not trigger
+            // the catch block below.
+            handleAADErrors(exchangeResponse);            
+        } 
+        else 
+        {
+            // For debugging:
+            // showMessage("ACCESS TOKEN: " + JSON.stringify(exchangeResponse.access_token));
+
+            // makeGraphApiCall makes an AJAX call to the MS Graph endpoint. Errors are caught
+            // in the .fail callback of that call, not in the catch block below.
+            makeGraphApiCall(exchangeResponse.access_token);
+        }
+    }
+    catch(exception) {
+        // The only exceptions caught here are exceptions in your code in the try block
+        // and errors returned from the call of `getAccessToken` above.
+        if (exception.code) { 
+            handleClientSideErrors(exception);
+        }
+        else {
+            showMessage("EXCEPTION: " + JSON.stringify(exception));
+        } 
+    }
+}
+
+async function getGraphToken(bootstrapToken) {
```

Comment:
This function does not need to be async and can just return a promise

Reply:
OK. Removed the async